src/Light.cpp

```diff
@@ -31,3 +31,13 @@ std::string Light::get_name() const
     std::string const name = typeid(decltype(*this)).name();
     return name.substr(6);
 }
+
+void Light::on_enabled()
+{
+    Renderer::get_instance()->register_light(std::static_pointer_cast<Light>(shared_from_this()));
```

Comment:
In the commit message it would be nice to say that we (un)register them in these methods.
Maybe `Light: (Un)register lights in on_enabled/disabled methods`